pkg/controller/build/imagebuilder/jobimagebuilder.go

```diff
@@ -101,6 +101,35 @@ func (j *jobImageBuilder) start(ctx context.Context) (*batchv1.Job, error) {
 	bj, err := j.kubeclient.BatchV1().Jobs(ctrlcommon.MCONamespace).Create(ctx, buildJob, metav1.CreateOptions{})
 	if err == nil {
 		klog.Infof("Build job %q created for MachineOSBuild %q", bj.Name, mosbName)
+
+		// Set the owner reference of the configmaps and secrets created to be the Job
+		// Set blockOwnerDeletion and Controller to false as Job ownership doesn't work when set to true
+		oref := metav1.NewControllerRef(bj, batchv1.SchemeGroupVersion.WithKind("Job"))
+		falseBool := false
+		oref.BlockOwnerDeletion = &falseBool
+		oref.Controller = &falseBool
+
+		cms, err := j.buildrequest.ConfigMaps()
+		if err != nil {
+			return nil, err
+		}
+		for _, cm := range cms {
+			cm.SetOwnerReferences([]metav1.OwnerReference{*oref})
```

Comment:
question (non-blocking): Can ownership be set when the object is created or does it need to occur as a post-creation update?

Reply:
It can be set when the object is created but the owner has to be created before we can set it on any other objects. For this case, the configmpas and secrets are created before the job is created that is why it is happening as a post create update here.

Reply:
Ahh, I understand now. Thanks :smile: